src/context/video-listing-context.js

```diff
@@ -0,0 +1,29 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import axios from "axios";
+
+const VideoListing = createContext();
+const useVideoListingContext = () => useContext(VideoListing);
+
+const VideoListingProvider = ({ children }) => {
+  const [videoList, setVideoList] = useState([]);
+
+  // getting videos data
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/videos");
```

Comment:
suggestion: you can write api calls functions a in separate file and import them wherever need

Reply:
I will do it later, thanks for suggesting.